x-pack/platform/plugins/shared/onechat/server/routes/agents.ts

```diff
@@ -6,96 +6,45 @@
-import { ServerSentEvent } from '@kbn/sse-utils';
-import { observableIntoEventSourceStream } from '@kbn/sse-utils-server';
-import { ChatAgentEvent } from '@kbn/onechat-common';
-import type { CallAgentResponse } from '../../common/http_api/agents';
+import { AgentType } from '@kbn/onechat-common';
+import type { ListAgentsResponse } from '../../common/http_api/agents';
 import { apiPrivileges } from '../../common/features';
+import { agentToDescriptor } from '../services/agents/utils';
 import type { RouteDependencies } from './types';
 import { getHandlerWrapper } from './wrap_handler';
 
 export function registerAgentRoutes({ router, getInternalServices, logger }: RouteDependencies) {
   const wrapHandler = getHandlerWrapper({ logger });
 
-  router.post(
+  router.get(
     {
-      path: '/internal/onechat/agents/invoke',
-      security: {
-        authz: { requiredPrivileges: [apiPrivileges.readOnechat] },
-      },
+      path: '/api/chat/agents',
       validate: {
-        body: schema.object({
-          agentId: schema.string({}),
-          agentParams: schema.recordOf(schema.string(), schema.any()),
+        query: schema.object({
+          type: schema.string({ defaultValue: AgentType.conversational }),
         }),
       },
-    },
-    wrapHandler(async (ctx, request, response) => {
-      const { agents } = getInternalServices();
-      const { agentId, agentParams } = request.body;
-      const agentRegistry = agents.registry.asPublicRegistry();
-
-      const agent = await agentRegistry.get({ agentId, request });
-      const agentResult = await agent.execute({
-        agentParams: agentParams as any,
-      });
-
-      return response.ok<CallAgentResponse>({
-        body: agentResult,
-      });
-    })
-  );
-
-  // stream agent events endpoint
-  router.post(
-    {
-      path: '/internal/onechat/agents/stream',
       security: {
         authz: { requiredPrivileges: [apiPrivileges.readOnechat] },
       },
-      validate: {
-        body: schema.object({
-          agentId: schema.string({}),
-          agentParams: schema.recordOf(schema.string(), schema.any()),
-        }),
+      options: {
+        availability: {
+          stability: 'experimental',
+        },
       },
     },
     wrapHandler(async (ctx, request, response) => {
-      const { agents } = getInternalServices();
-      const { agentId, agentParams } = request.body;
-      const agentRegistry = agents.registry.asPublicRegistry();
+      // const { type } = request.query;
+      const { agents: agentService } = getInternalServices();
+      const agentRegistry = agentService.registry.asPublicRegistry();
 
-      const abortController = new AbortController();
-      request.events.aborted$.subscribe(() => {
-        abortController.abort();
-      });
-
-      const subject$ = new Subject<ChatAgentEvent>();
+      // TODO: use type as filter?
+      const agents = await agentRegistry.list({ request });
```

Comment:
I struggle to understand a difference between agents vs agent profiles. I was not as involved in the conversation leading to splitting this up.... but I understand that eventually we will unify this under `/agent` route

Reply:
That's correct. It wasn't clear until very recently what "agents" were, and what those "custom agents feature" we're building was going to be named. Now we know - those are agents - so we will be making the APIs more consistent. There is a few things to consider though, so I'm keeping this for later.